#!/usr/bin/env python
PACKAGE = "rrg_nbv_exploration"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("distance_factor", double_t, 0, "Factor to adjust the weight of distance to the robot for the reward function.",    1.0, 0,   1)
gen.add("traversability_factor", double_t, 0, "Factor to adjust the weight of traversability evaluation for the reward function.",    1.0, 0,   1)
gen.add("heading_factor", double_t, 0, "Factor to adjust the weight of heading change to the robot for the reward function.",    1.0, 0,   1)
gen.add("radius_factor", double_t, 0, "Factor to adjust the weight of the node's radius for the reward function.",    1.0, 0,   1)
gen.add("local_sampling_radius", double_t, 0, "Radius in m around the robot in which local samples are placed.",    5.0, 0,   100)
gen.add("local_graph_radius", double_t, 0, "Radius in m around the robot in which the local RRG is built and maintained.",    10.0, 0,   500)


exit(gen.generate(PACKAGE, "rneNode", "GraphConstructor"))